gemrb/plugins/SDLVideo/SDLSurfaceSprite2D.cpp

```diff
@@ -178,8 +178,10 @@ void SDLSurfaceSprite2D::EnsureShadedPalette() const noexcept
 void SDLSurfaceSprite2D::ShadePalette(BlitFlags renderflags, const Color* tint) const noexcept
 {
 	Palette::Colors buffer;
+	buffer[0] = format.palette->GetColorAt(0);
 
-	for (size_t i = 1; i < 256; ++i) {
+	size_t startIndex = format.HasColorKey ? 1 : 0;
+	for (size_t i = startIndex; i < 256; ++i) {
 		buffer[i] = format.palette->GetColorAt(i);
```

Comment:
I seems, that this can simply be written as 
```cpp
for (size_t i = 0; i < 256; ++i) {
```

Reply:
No, the loop performs tinting and stuff. In case of color key, index 0 must not be touched.